alf/utils/decoding_network.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gin.tf
+import tensorflow as tf
+
+
+@gin.configurable
+class DecodingNetwork(tf.keras.models.Model):
```

Comment:
ImageDecodingNetwork to be precise.

Reply:
This can be a decoding network for floating vector if conv_layer_params is None.

Reply:
renamed.